app/lib/ingestors/google/docs.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class Ingestors::Google::Docs
+  def initialize(user)
+    client = Signet::OAuth2::Client.new(
+      token_credential_uri: 'https://oauth2.googleapis.com/token',
+      client_id: ENV.fetch('GOOGLE_CLIENT_ID', nil),
+      client_secret: ENV.fetch('GOOGLE_CLIENT_SECRET', nil),
+      refresh_token: user.google_oauth_credentials['refresh_token']
+    )
+    client.fetch_access_token!
+
+    @client = client
+  end
+
+  def ingest
+    drive = Google::Apis::DriveV2::DriveService.new
+    drive.authorization = client
+
+    files = drive.list_files(q: 'mimeType = "application/vnd.google-apps.document"').items
+    files.each(&method(:persist_or_update_doc))
```

Comment:
Is there a way for us to only update "dirty" files? I think there might be significant overhead otherwise

Reply:
Yes, I think it would involve tracking the 'last_sync_time' in the DB. Added a TODO to add this in the next revision.